client/client.go

```diff
@@ -360,7 +362,14 @@ func (r *NotaryRepository) AddTarget(target *Target, roles ...data.RoleName) err
 	}
 	logrus.Debugf("Adding target \"%s\" with sha256 \"%x\" and size %d bytes.\n", target.Name, target.Hashes["sha256"], target.Length)
 
-	meta := data.FileMeta{Length: target.Length, Hashes: target.Hashes}
+	var customData *canonicaljson.RawMessage
+	if target.Custom != nil {
+		customData = new(canonicaljson.RawMessage)
+		if err := customData.UnmarshalJSON(target.Custom); err != nil {
```

Comment:
Is this just to validating that `Target.Custom` is actually valid `json` data and not a cast `string`?